### ISSUE TITLE
Checksum fails for different images from different mirrors

### ISSUE THREAD

**Issue by wyq977** [docker/for-mac#4654]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 4788C2C7-56B8-4DAA-AEDF-93C229702E25/20200608094306

### Expected behavior

Pulling the image or build the image

### Actual behavior

```shell
# different images have different sha256 value
filesystem layer verification failed for digest sha256: 
```

### Information

  - macOS Version: 10.15.5

### Diagnostic logs

```
Docker for Mac: version...

```

### Steps to reproduce the behavior

  1. `docker pull ubuntu` or any other public images

Also try mirrors available in China such as https://hub-mirror.c.163.com and http://f1361db2.m.daocloud.io 

Also try `docker load -i ~/ubuntu.tar` to make sure the internet is not a issue but also fails with the same error.

Thinking it might be related to the drive in macOS Catalina but nothing similar comes up in my search. I am using a WD SN550 512GB NVME SSD in AFPS format (hackintosh ) but not sure if that is related.

Something that might be related but solved is #2625 and it seems to me that there is no option for users to choose between raw and qcow.

**Comment by wyq977** [comment]
I managed to pull the alpine image and tried the docker image in #2625 once and the text displayed fine, indicating that the disk/image corruption might not be the root of this behavior. 

Will try to eliminate problems arising from unstable Internet connection

**Comment by wyq977** [comment]
Yeah I am pretty sure now that internet is not an issue...

**Comment by davisinfo** [comment]
Happens for me also ...

**Comment by lucaslg26** [comment]
@wyq977 did you found any solution? I have a similar setup and I'm stuck with same error even with a HFS drive.

**Comment by gengpei** [comment]
wasted my 2 days, solved by remove Cpuid1Data and Cpuid1Mask's data. using Catalina 10.15.5 and 10th desktop cpu, this version macos support 10th, so don't need Cpuid1Data any more.

**Comment by lucaslg26** [comment]
> wasted my 2 days, solved by remove Cpuid1Data and Cpuid1Mask's data. using Catalina 10.15.5 and 10th desktop cpu, this version macos support 10th, so don't need Cpuid1Data any more.

I just need try to solve this problem with a unsurpotted CPU. :cry:

**Comment by davisinfo** [comment]
Cpuid1Data was the problem for me also, probably it activates some instruction sets/features that are not actually supported by the CPU.

**Comment by davisinfo** [comment]
> > wasted my 2 days, solved by remove Cpuid1Data and Cpuid1Mask's data. using Catalina 10.15.5 and 10th desktop cpu, this version macos support 10th, so don't need Cpuid1Data any more.
> 
> I just need try to solve this problem with a unsurpotted CPU. 😢

What is your CPU?

**Comment by lucaslg26** [comment]
> > > wasted my 2 days, solved by remove Cpuid1Data and Cpuid1Mask's data. using Catalina 10.15.5 and 10th desktop cpu, this version macos support 10th, so don't need Cpuid1Data any more.
> > 
> > 
> > I just need try to solve this problem with a unsurpotted CPU. 😢
> 
> What is your CPU?

My CPU is a Haswell-EP(Xeon E5 2660v3) and I'm using cpuid1data from Haswell

**Comment by lucaslg26** [comment]
Wow.
Removed cpuid1data and cpu1Mask and worked for me too! 
But unsurprisingly CPU Power Management isnt working anymore.

**Comment by davisinfo** [comment]
> Wow.
> Remove cpuid1data and cpu1Mask worked for me too!
> But unsurprisingly CPU Power Management isnt working anymore.

You can add the xcpm bootstrap patch and power management will work. What is very strange is that even if I set the actual CPUID that the CPU has it still breaks so it looks like an OpenCore issue.

**Comment by lucaslg26** [comment]
> > Wow.
> > Remove cpuid1data and cpu1Mask worked for me too!
> > But unsurprisingly CPU Power Management isnt working anymore.
> 
> You can add the xcpm bootstrap patch and power management will work. What is very strange is that even if I set the actual CPUID that the CPU has it still breaks so it looks like an OpenCore issue.

Is there a quirk in opencore?

I will try later

**Comment by davisinfo** [comment]
No, it is a kernel patch. See https://www.insanelymac.com/forum/topic/341477-open-core-kernel-kext-patch-for-x99x299-motherboard/?page=5

**Comment by lucaslg26** [comment]
> No, it is a kernel patch. See https://www.insanelymac.com/forum/topic/341477-open-core-kernel-kext-patch-for-x99x299-motherboard/?page=5

it works!

Thanks!

**Comment by yizhiheng** [comment]
@gengpei Your solution works for me, thanks!

**Comment by obarlev** [comment]
> wasted my 2 days, solved by remove Cpuid1Data and Cpuid1Mask's data. using Catalina 10.15.5 and 10th desktop cpu, this version macos support 10th, so don't need Cpuid1Data any more.

You're a life saver! 
I ended up removing the Cpuid1Data and Cpuid1Mask's. Per intel power gadget, seems like the power management works.
My machine is 10900K based.
Is there any impact / tradeoff for keeping the Cpuid1Data and Cpuid1Mask's as empty?

**Comment by davisinfo** [comment]
No, you only need to set them if your CPU is unsupported.

**Comment by wyq977** [comment]
For details you can try something mention here https://github.com/acidanthera/bugtracker/issues/1035.
This is something related to the wrong microcode version caused by CPUID in OC loader
